objects/src/notes/metadata.rs

```diff
@@ -19,12 +19,39 @@ pub struct NoteMetadata {
     sender: AccountId,
     tag: Felt,
     note_type: NoteType,
+    aux: Felt,
 }
```

Comment:
I don't understand, why are we changing it to u32?

added comments

Reply:
I had an explanation of this in https://github.com/0xPolygonMiden/miden-base/issues/402#issuecomment-2008397166 but briefly: this allows us to avoid tag prefixes - we always query on the full tag which is now 32 bits.

Reply:
and how do you want this to look? this will affect all the bit manipulations. should the 32 bits be in the most or least significant part of the word?

Reply:
I'm going to pack this in the least significant bytes, I hope this won't cause confusion when updating the node and client.

As a follow up, why is this being done? I see zero benefits here

Reply:
Yes, I think putting tag into the 32 least significant of the element (not word) is fine.

As for motivation: i'm trying to make the tag as light-weight as possible for a couple of reasons:
1. It is a prat of public data associated with each note - so, needs to be stored in the node database.
  a. Related to this, we want to make sure that we can compute `hash(note_id || note_metadata)` in a single permutation of the hash function, and this means that all metadata needs to fit into a single word.
2. It is something that we'll need to index on to serve state sync and other tag-based requests.
3. The client may need to send many tags to the node during state sync request - so, having smaller tags reduces request size.

Reply:
changed it to `u32`